nodejs/eks/cluster.ts

```diff
@@ -627,6 +624,76 @@ export function createCore(
+        const kubeProxyVersion: pulumi.Output<string> = args.kubeProxyAddonOptions?.version
+            ? pulumi.output(args.kubeProxyAddonOptions?.version)
+            : aws.eks
+                  .getAddonVersionOutput(
+                      {
+                          addonName: "kube-proxy",
+                          kubernetesVersion: eksCluster.version,
+                          mostRecent: true, // whether to return the default version or the most recent version for the specified kubernetes version
+                      },
+                      { parent, provider },
+                  )
+                  .apply((addonVersion) => addonVersion.version);
+
+        const kubeProxyAddon = new aws.eks.Addon(
+            `${name}-kube-proxy`,
+            {
+                clusterName: eksCluster.name,
+                addonName: "kube-proxy",
+                preserve: true,
+                tags: args.tags,
+                resolveConflictsOnCreate:
+                    args.kubeProxyAddonOptions?.resolveConflictsOnCreate ?? "OVERWRITE",
+                resolveConflictsOnUpdate:
+                    args.kubeProxyAddonOptions?.resolveConflictsOnUpdate ?? "OVERWRITE",
+                addonVersion: kubeProxyVersion,
+            },
+            { parent, provider },
+        );
+    }
+
+    if (args.corednsAddonOptions?.enabled ?? true) {
+        const corednsVersion: pulumi.Output<string> = args.corednsAddonOptions?.version
+            ? pulumi.output(args.corednsAddonOptions.version)
+            : aws.eks
+                  .getAddonVersionOutput(
+                      {
+                          addonName: "coredns",
+                          kubernetesVersion: eksCluster.version,
+                          mostRecent: true, // whether to return the default version or the most recent version for the specified kubernetes version
+                      },
+                      { parent, provider },
+                  )
+                  .apply((addonVersion) => addonVersion.version);
+
+        const corednsAddon = new aws.eks.Addon(
+            `${name}-coredns`,
+            {
+                clusterName: eksCluster.name,
+                addonName: "coredns",
+                tags: args.tags,
+                preserve: true,
+                addonVersion: corednsVersion,
+                resolveConflictsOnCreate:
+                    args.corednsAddonOptions?.resolveConflictsOnCreate ?? "OVERWRITE",
+                resolveConflictsOnUpdate:
+                    args.corednsAddonOptions?.resolveConflictsOnUpdate ?? "OVERWRITE",
+                configurationValues: pulumi.output(args.fargate).apply((fargate) => {
+                    if (fargate) {
+                        return JSON.stringify({
+                            computeType: "Fargate",
+                        });
+                    }
+                    return undefined;
+                }) as any,
```

Comment:
Nit: disabling type-checking via any here avoids some pain perhaps?

Reply:
`any` is gonna get removed here when we expose the `configurationValues`. That's a follow up item.

Reply:
https://github.com/pulumi/pulumi-eks/issues/1369

Reply:
The referenced issue is https://github.com/pulumi/pulumi/issues/6175